bioio/tests/test_bio_image.py

```diff
@@ -72,37 +71,189 @@ def test_bioimage_submission_data_reader_type_alignment(
 
 
 def test_bioimage_attempts_s3_read_with_anon_attr(
-    sample_text_file: pathlib.Path, dummy_plugin: str
+    sample_text_file: pathlib.Path,
+    plugin_factory: Callable[[Iterable[TestPluginSpec]], list[EntryPoint]],
```

Comment:
I might be naive about how to write Python tests, but the code looks weird having plugin_factory be an input argument to all the functions.  A type definition to shorten `Callable[[Iterable[TestWriterSpec]], list[EntryPoint]]` would probably be nice for readability.  Maybe name it as `plugin_factory: PluginFactoryFixture` ?

Reply:
added in [3d3f727](https://github.com/bioio-devs/bioio/pull/162/commits/3d3f727d37d7febaeca58891bacaea6c6024a7d5) Type definition doesn't quite work without some aliasing so I added some protocol support should work the same way